setup.py

```diff
@@ -22,40 +22,63 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio module that adds filesystem abstraction."""
+"""Invenio module for document management."""
 
 import os
 import sys
 
-from setuptools import setup
+from setuptools import find_packages, setup
 from setuptools.command.test import test as TestCommand
 
 readme = open('README.rst').read()
 history = open('CHANGES.rst').read()
 
-requirements = [
-    'Flask>=0.10.1',
-    'fs>=0.4',
-    'invenio-base>=0.1.0',
-    'invenio-ext>=0.2.1',
-    'invenio-utils>=0.1.1',
-    'jsonpatch>=1.11',
-    'jsonschema>=2.5.1',
-    'six>=1.7.2',
-]
-
-test_requirements = [
-    'unittest2>=1.1.0',
-    'Flask_Testing>=0.4.1',
-    'pytest>=2.7.0',
+tests_require = [
+    'check-manifest>=0.25',
+    'coverage>=4.0',
+    'isort>=4.2.2',
+    'pep257>=0.7.0',
+    'pytest-cache>=1.0',
     'pytest-cov>=1.8.0',
     'pytest-pep8>=1.0.6',
-    'coverage>=3.7.1',
+    'pytest>=2.8.0',
 ]
 
+extras_require = {
+    'docs': [
+        "Sphinx>=1.3",
+    ],
+    'mysql': [
+        'invenio-records[mysql]>=1.0.0a2',
+    ],
+    'postgresql': [
+        'invenio-records[postgresql]>=1.0.0a2',
+    ],
+    'tests': tests_require,
+}
+
+extras_require['all'] = []
+for reqs in extras_require.values():
+    extras_require['all'].extend(reqs)
 
-class PyTest(TestCommand):
+setup_requires = [
+    'Babel>=1.3',
+]
+
+install_requires = [
+    'Flask-BabelEx>=0.9.2',
+    'Flask-CLI>=0.2.1',
+    'fs>=0.5.3',
+    'invenio-db>=1.0.0a4',
+    'invenio-records>=1.0.0a2',
+    'jsonpointer>=1.9',
```

Comment:
As discussed IRL, it's OK to depend on `invenio-records`, though for later use cases (e.g. CDS commenting system that is accepting full-text document files) we may perhaps revive the notion of standalone documents.  

(This was originally introduced for INSPIRE plots's needs, but I think this is no longer required in the optics of "everybody is a first class citizen" files included.)